### PR TITLE
Add account_payment_order_transfer_journal.

### PR DESCRIPTION
**Pull request by adrienpeiffer**
This module allow to generate payment order entries on a transfer journal.

Add fonctionnality removed in #822

### CODE REVIEW

File: account_payment_order_transfer_journal/models/account_journal.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+
+class AccountJournal(models.Model):
+
+    _inherit = "account.journal"
+
+    transfer_journal_id = fields.Many2one(
+        comodel_name="account.journal",
+        string="Transfer journal on payment/debit orders",
+        help="Journal to write payment entries when confirming " "payment/debit orders",
```
Comment (acsonefho) [inline account_payment_order_transfer_journal/models/account_journal.py]: Mal-formed string

File: account_payment_order_transfer_journal/tests/test_payment_order.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                "name": "Test account",
+                "code": "TEST1",
+                "user_type_id": self.env.ref("account.data_account_type_expenses").id,
+            }
+        )
+        self.bank_journal = (
+            self.env["account.journal"].search(
+                [("company_id", "=", self.company.id), ("type", "=", "bank")], limit=1
+            ),
+        )[0]
```
Comment (acsonefho) [inline account_payment_order_transfer_journal/tests/test_payment_order.py]: `[0]` is useless

File: account_payment_order_transfer_journal/tests/test_payment_order.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        self.transfer_journal = (
+            self.env["account.journal"].search(
+                [("company_id", "=", self.company.id), ("type", "=", "general")],
+                limit=1,
+            ),
+        )[0]
```
Comment (acsonefho) [inline account_payment_order_transfer_journal/tests/test_payment_order.py]: `[0]` is useless

File: account_payment_order_transfer_journal/tests/test_payment_order.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+    def test_payment_order_transfer_jounral(self):
+        self.invoice.action_post()
+        self.bank_journal.transfer_journal_id = self.transfer_journal
+        order_vals = {
+            "payment_type": "outbound",
+            "payment_mode_id": self.mode.id,
+        }
+        order = self.env["account.payment.order"].create(order_vals)
+        with self.assertRaises(UserError):
```
Comment (acsonefho) [inline account_payment_order_transfer_journal/tests/test_payment_order.py]: You should check if the exception raised is the expected one
```
with self.assertRaises(UserError) as em:
    order.draft2open()
self.assertIn("A fragment message", em.exception.args[0])
```

File: account_payment_order_transfer_journal/tests/test_payment_order.py [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+            order.draft2open()
+
+        order.payment_mode_id = self.mode.id
+        order.payment_mode_id_change()
+        self.assertEqual(order.journal_id.id, self.bank_journal.id)
+
+        self.assertEqual(len(order.payment_line_ids), 0)
+        with self.assertRaises(UserError):
+            order.draft2open()
```
Comment (acsonefho) [inline account_payment_order_transfer_journal/tests/test_payment_order.py]: Same here

**Review by acsonefho — CHANGES_REQUESTED** [review]
Code LGTM.
Minor changes

**Comment by adrienpeiffer** [comment]
@acsonefho :rocket:

File: account_payment_order_transfer_journal/tests/test_payment_order.py (hunk 4 above)
Comment (acsonefho) [reply account_payment_order_transfer_journal/tests/test_payment_order.py]: Why did you finally remove these 2 lines? Not relevant?

Comment (adrienpeiffer) [reply account_payment_order_transfer_journal/tests/test_payment_order.py]: Indeed. Already tested in account_payment_order

**Review by acsonefho — APPROVED** [review]

**Review by luc-demeyer — APPROVED** [review]

**Comment by luc-demeyer** [comment]
Code review & testing ok.

**Comment by qgroulard** [comment]
Won't work since https://github.com/OCA/bank-payment/pull/979 unfortunately.
You'll need to rebase and adapt code.

**Pull request closed without merging**